client/clientBean.go

```diff
@@ -78,36 +78,55 @@ func NewClientBean(factory Factory, clusterMetadata cluster.Metadata) (Bean, err
 		return nil, err
 	}
 
-	remoteAdminClients := map[string]adminservice.AdminServiceClient{}
-	remoteFrontendClients := map[string]workflowservice.WorkflowServiceClient{}
+	adminClients := map[string]adminservice.AdminServiceClient{}
+	frontendClients := map[string]workflowservice.WorkflowServiceClient{}
 
+	currentClusterName := clusterMetadata.GetCurrentClusterName()
 	for clusterName, info := range clusterMetadata.GetAllClusterInfo() {
 		if !info.Enabled {
 			continue
 		}
 
-		adminClient := factory.NewAdminClientWithTimeout(
-			info.RPCAddress,
-			admin.DefaultTimeout,
-			admin.DefaultLargeTimeout,
-		)
-
-		remoteFrontendClient := factory.NewFrontendClientWithTimeout(
-			info.RPCAddress,
-			frontend.DefaultTimeout,
-			frontend.DefaultLongPollTimeout,
-		)
-
-		remoteAdminClients[clusterName] = adminClient
-		remoteFrontendClients[clusterName] = remoteFrontendClient
+		var adminClient adminservice.AdminServiceClient
+		var frontendClient workflowservice.WorkflowServiceClient
+		if clusterName != currentClusterName {
+			adminClient = factory.NewAdminClientWithTimeout(
+				info.RPCAddress,
+				admin.DefaultTimeout,
+				admin.DefaultLargeTimeout,
+			)
+			frontendClient = factory.NewFrontendClientWithTimeout(
+				info.RPCAddress,
+				frontend.DefaultTimeout,
+				frontend.DefaultLongPollTimeout,
+			)
+		} else {
+			// Init inner cluster client with membership info
+			adminClient, err = factory.NewLocalAdminClientWithTimeout(
```

Comment:
should we move this block outside of this for loop so that local client will always be initialized? currently client creation may skip because of check at line 86.

Reply:
Updated